src/main/java/org/terasology/metalrenegades/ai/component/NeedsComponent.java

```diff
@@ -26,6 +27,27 @@
  */
 public class NeedsComponent implements Component {
 
-    public Map<CitizenNeed.Type, CitizenNeed> needs = new EnumMap<>(CitizenNeed.Type.class);
+    @Replicate
+    public CitizenNeed hungerNeed;
```

Comment:
The only downside that I can see with this approach is that it is not extensible. There are now multiple files that need to be modified if someone wants to add a new need in the future. 

As most need calculations will be done on the server, and it's just the tooltip that's on the client, you could keep the map and also store an array of strings, replicate just that array and print it out as it is in the tooltip.
- Pros:
  - Easier to add/change needs
- Cons:
  - That entire array has to be modified whenever a need is changed
  - This array only exists to display tooltips and cannot be used anywhere else

Can discuss this further, let me know what you think

Reply:
Keeping the citizen need map approach would probably be fine if the only problem was multiplayer replication, but those replication issues stem from the component not serializing properly. An unserializable field causes three different issues:

- Needs status doesn't replicate over the server properly (the main topic of this PR)
- Needs behaviors also break whenever the character is unloaded (if a player walks away from a character then walks back, the needs are completely broken, and the console is filled with NPEs)
- Needs status won't save when the game is closed, leading to the same broken behavior and NPEs as above when the game is loaded again

If we can find a collection form to store needs in that can still be serialized then that would probably be the best way to do this 😄